ocf_datapipes/config/model.py

```diff
@@ -127,6 +127,48 @@ def history_seq_length_60_minutes(self):
         return int(np.ceil(self.history_minutes / 60))
 
 
+class DropoutMixin(Base):
+    """Mixin class, to add dropout minutes"""
+
+    dropout_timedeltas_minutes: List[int] = Field(
+        None,
+        description="List of possible minutes before t0 where data availability may start. Must be "
+        "negative or zero.",
+    )
+
+    dropout_fraction: int = Field(0, description="Chance of dropout being applied to each sample")
+
+    @validator("dropout_timedeltas_minutes")
+    def dropout_timedeltas_minutes_negative(cls, v):
+        """Validate 'dropout_timedeltas_minutes'"""
+        if v is not None:
+            for m in v:
+                assert m <= 0
+        return v
+
+    @validator("dropout_fraction")
+    def dropout_fraction_valid(cls, v):
+        """Validate 'dropout_fraction'"""
+        assert 0 <= v <= 1
+        return v
+
+
+class SystemDropoutMixin(Base):
+    """Mixin class, to add independent system dropout"""
+
+    system_dropout_timedeltas_minutes: List[int] = Field(
+        None,
+        description="List of possible minutes before t0 where data availability may start. Must be "
+        "negative or zero. Each system in a sample is delayed independently from the other by "
+        "values randomly selected from this list.",
+    )
+
+    # The degree of system dropout for each returned sample will be randomly drawn from
+    # the range [system_dropout_fraction_min, system_dropout_fraction_max]
+    system_dropout_fraction_min: int = Field(0, description="Min chance of system dropout")
```

Comment:
Why is a chance an int? Should this be float?

Reply:
Oh yeh, definitely should be a float!